lib/y2logsstep.pm

```diff
@@ -107,6 +107,81 @@ sub check_and_record_dependency_problems {
     }
 }
 
+# if dependency issues appear during installation, break dependency and continue with installation
+sub break_dependency {
+    my ($self) = @_;
+    return unless check_screen("inst-overview-dep-warning", 1);
+
+    if (check_var('VIDEOMODE', 'text')) {
+        send_key 'alt-c';    # Change
+        assert_screen 'inst-overview-options';
+        send_key 'alt-s';    # Software
+
+      LOOP: do {
+            send_key 'alt-s';    # Solution
+            if (check_screen 'break_not_highlighted') {
+                send_key 'down';    # down to option break dependency
+            }
+            send_key 'ret';         # select option break dependency
+            send_key 'alt-o';       # OK - Try Again
+        } while (check_screen 'package_dependency', 5);    # repeat it untill all dependency issues are resolved
+
+        if (check_screen 'installation_summary') {
+            send_key 'alt-a';                              # Accept
+        }
+        if (check_screen 'licence_agreement') {
+            send_key 'alt-a';                              # Accept
+        }
+        if (check_screen 'automatic_changes') {
+            send_key 'alt-o';                              # OK
+        }
+        if (check_screen 'installtion_overview') {
+            send_key 'alt-i';                              # Install
+        }
+        if (check_screen 'confirm_installation') {
+            send_key 'alt-i';                              # Install
+        }
+        if (check_screen 'error_before_install_text') {
+            send_key 'alt-o';                              # OK
+            send_key 'alt-s';                              # Solution
+            send_key 'down';                               # go to the similar option like break
+            send_key 'ret';
+            send_key 'alt-o', 1;                           # OK -- Try Again
+        }
+        goto LOOP;                                         # repeat selections again
+    }
+
+    else {
+        send_key_until_needlematch 'packages-section-selected', 'tab';
+        send_key 'ret';
+        assert_screen 'dependancy-issue';                  # make sure the dependancy issue is actually showing
+
+      LOOP: do {
+            wait_screen_change {
+                send_key 'alt-2';                          # 2 is the option to break dependency
+            }
+            wait_screen_change {
+                send_key 'spc';                            # select it
+            }
+            send_key 'alt-o';                              # OK
+        } while (check_screen 'dependancy-issue', 5);
+
+        wait_screen_change {
+            send_key 'alt-a';                              # Accept
+        }
+        send_key 'alt-a', 3;                               # Accept
```

Comment:
Please avoid the second argument on send_key. It's a boolean value (so only 0 or 1 should be used) but internally it calls wait_idle which we like to avoid